src/goto-synthesizer/cegis_verifier.cpp

```diff
@@ -53,6 +53,55 @@ static bool contains_symbol_prefix(const exprt &expr, const std::string &prefix)
   return false;
 }
 
+static const exprt &
+get_checked_pointer_from_null_pointer_check(const exprt &violation)
+{
+  // A NULL-pointer check is the negation of an equation between the checked
+  // pointer and a NULL pointer.
+  const exprt *equal_expr = &violation;
+
+  // Extract the equal expression in the violation.
+  while(equal_expr->id() != ID_equal)
+  {
+    INVARIANT(equal_expr->operands().size() > 0, " No equal expression found.");
+    equal_expr = &equal_expr->operands()[0];
+  }
+
+  const exprt *lhs_pointer = &equal_expr->operands()[0];
+  const exprt *rhs_pointer = &equal_expr->operands()[1];
```

Comment:
If my above code suggestion works for you then this should be `const exprt &lhs_pointer = equal_expr.lhs();` and `const exprt &rhs_pointer = equal_expr.rhs();`